### PR TITLE
Add getCapacity to Mempool API

### LINKED ISSUE #1510: Public mempool API to query the size limit
End users would benefit from obtaining information regarding the limit of the mempool, such that, for example, the `MempoolSize` (and, hence `MempoolSnapshot`) could be extended to contain not only the its current byte/Tx occupancy, but also the byte limit:

```haskell
-- | The size of a mempool.
data MempoolSize = MempoolSize
  { msNumTxs   :: !Word32
    -- ^ The number of transactions in the mempool.
  , msNumBytes :: !Word32
    -- ^ The summed byte size of all the transactions in the mempool.
  , msMaxBytes :: !Word32
    -- ^ The upper limit on the byte size of the mempool.
  } deriving (Eq, Show)
```

### PR DESCRIPTION
**Pull request by intricate**
Closes #1510

### CODE REVIEW

**Comment by mrBliss** [comment]
(Oops, I thought I had submitted my review last week, but I forgot to click the submit button :man_facepalming:)

File: ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs [hunk 1]
```diff
@@ -269,24 +278,27 @@ data BlockSlot =
...
     -- number, if it exists.
-  , snapshotLookupTx    :: idx -> Maybe (GenTx blk)
+  , snapshotLookupTx        :: idx -> Maybe (GenTx blk)
 
     -- | Get the size of the mempool snapshot.
-  , snapshotMempoolSize :: MempoolSize
+  , snapshotMempoolSize     :: MempoolSize
+
+    -- | Get the capacity of the mempool snapshot.
+  , snapshotMempoolCapacity :: MempoolCapacityBytes
```
[the remark this answers is not in the document]
Comment (intricate) [reply ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs]: I sent @deepfire a message about this, so I'll know which one to drop soon.

[the remark this answers is not in the document]
Comment (deepfire) [reply ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs]: @intricate, @mrBliss, I have nothing to sway me in either direction (that I can think of), so I'll defer to your judgement as to what would make more sense in the long run.

File: ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs [hunk 2]
```diff
@@ -442,6 +437,13 @@ implGetSnapshotFor MempoolEnv{mpEnvStateVar, mpEnvLedgerCfg}
         . internalStateFromVR
         . validateStateFor mpEnvLedgerCfg blockSlot ledger
 
+-- TODO #1498:
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs]: Can you make this a proper haddock docstring:

```haskell
-- | Return the current capacity of the Mempool.
--
-- TODO #1498: As of right now, ...
```

Comment (intricate) [reply ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs]: Done

**Pull request merged**